core/src/OpenACC/Kokkos_OpenACC_Instance.cpp

```diff
@@ -58,31 +58,38 @@ namespace Kokkos {
 bool show_warnings() noexcept;
 }
 
-Kokkos::Experimental::Impl::OpenACCInternal*
+Kokkos::Experimental::Impl::OpenACCInternal&
 Kokkos::Experimental::Impl::OpenACCInternal::singleton() {
   static OpenACCInternal self;
-  return &self;
+  return self;
 }
 
-void Kokkos::Experimental::Impl::OpenACCInternal::initialize(
-    InitializationSettings const& settings) {
-  if (OpenACC_Traits::may_fallback_to_host &&
-      acc_get_num_devices(OpenACC_Traits::dev_type) == 0 &&
-      !settings.has_device_id()) {
-    if (show_warnings()) {
-      std::cerr << "Warning: No GPU available for execution, falling back to"
-                   " using the host!"
-                << std::endl;
-    }
-    acc_set_device_type(acc_device_host);
-    // FIXME_OPENACC if multiple execution space instances are supported,
-    // device id variable should be explicitly set to the value returned by
-    // acc_get_device_num(acc_device_host).
-  } else {
-    using Kokkos::Impl::get_gpu;
-    int const dev_num = get_gpu(settings);
-    acc_set_device_num(dev_num, OpenACC_Traits::dev_type);
+int Kokkos::Experimental::Impl::OpenACCInternal::verify_is_initialized(
+    const char* const label) const {
+  if (m_accDev < 0) {
```

Comment:
I would only check that `m_is_initialized` is `true`.

Reply:
Updated.